parity-clib/examples/cpp/main.cpp

```diff
@@ -50,136 +51,141 @@ const std::vector<std::string> ws_subscriptions {
 };
 
 // callback that gets invoked upon an event
-void callback(void* user_data, const char* response, size_t _len) {
-	Callback* cb = static_cast<Callback*>(user_data);
-	if (cb->type == CALLBACK_RPC) {
-		cb->counter -= 1;
-	} else if (cb->type == CALLBACK_WS) {
-		std::regex is_subscription ("\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-F0-9]{16}\",\"id\":1\\}");
-		if (std::regex_match(response, is_subscription) == true) {
-			cb->counter -= 1;
-		}
-	}
+void callback(void *user_data, const char *response, size_t _len) {
+  Callback *cb = static_cast<Callback *>(user_data);
+  if (cb->type == CALLBACK_RPC) {
+    cb->counter -= 1;
+  } else if (cb->type == CALLBACK_WS) {
+    std::regex is_subscription("\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-"
+                               "F0-9]{16}\",\"id\":1\\}");
+    if (std::regex_match(response, is_subscription)) {
+      cb->counter -= 1;
+    }
+  }
 }
 
 int main() {
-	// run full-client
-	{
-		std::vector<const char*> config = {"--no-ipc" , "--jsonrpc-apis=all", "--chain", "kovan"};
-		void* parity = parity_run(config);
-		if (parity_rpc_queries(parity)) {
-			printf("rpc_queries failed\r\n");
-			return 1;
-		}
-
-		if (parity_subscribe_to_websocket(parity)) {
-			printf("ws_queries failed\r\n");
-			return 1;
-		}
-
-		if (parity != nullptr) {
-			parity_destroy(parity);
-		}
-	}
-
-	// run light-client
-	{
-		std::vector<const char*> light_config = {"--no-ipc", "--light", "--jsonrpc-apis=all", "--chain", "kovan"};
-		void* parity = parity_run(light_config);
-
-		if (parity_rpc_queries(parity)) {
-			printf("rpc_queries failed\r\n");
-			return 1;
-		}
-
-		if (parity_subscribe_to_websocket(parity)) {
-			printf("ws_queries failed\r\n");
-			return 1;
-		}
-
-		if (parity != nullptr) {
-			parity_destroy(parity);
-		}
-	}
-	return 0;
+  // run full-client
+  {
+    std::vector<const char *> config = {"--no-ipc", "--jsonrpc-apis=all",
+                                        "--chain", "kovan"};
+    struct parity_ethereum *parity = parity_run(config);
+    if (parity_rpc_queries(parity)) {
+      printf("rpc_queries failed\r\n");
```

Comment:
Not sure if done here but make sure to use C++'s version of `printf` (found in <cstdio>) here instead of C's version.